### ISSUE TITLE
running backup gives a 429 error

### ISSUE THREAD

**Issue by wtfrank** [tedivm/screeps-backup#2]
requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://screeps.com/api/user/memory-segment?segment=25&shard=shard1

**Comment by AlinaNova21** [comment]
You are likely hitting rate limiting, make sure you aren't making more than a couple requests per second.

**Comment by wtfrank** [comment]
That's exactly what is happening - when backup is working on memory segments it downloads potentially 100 segments per shard, one after the other.

**Comment by tedivm** [comment]
Yup! The ratelimiting totally broke this program, and I don't care enough to fix it anymore. Your options are to run it manually after hitting the button to disable ratelimiting in the Screeps web client, or to help rewrite the program to reduce the number of calls (in theory they modified the API so you can request more than one segment at a time, so that may work).

**Comment by AlinaNova21** [comment]
`screeps-api` on npm now has a cli tool, this can be used for downloading and setting memory, segments, and code straight from the CLI. https://github.com/screepers/node-screeps-api#cli-usage :) It also allows for downloading all 100 segments in a single request with the previously mentioned multiple segment requests :)
